### PR TITLE
Enable clang-format scan in public CI

### PR DESCRIPTION
**Pull request by napetrov**
# Description
With this change we enable regular validation with clang-format

Changes proposed in this pull request:
- enable automation for clang-format validation
- apply clang-format on existing codebase

### CODE REVIEW

**Comment by napetrov** [comment]
make test

**Comment by napetrov** [comment]
make test

File: .ci/scripts/clang-format.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+
+echo "Using clang-format version: $(clang-format --version)"
+echo "Starting format check..."
+
+for filename in $(find . -type f | grep -P ".*\.(c|cpp|h|hpp|cl)$"); do clang-format -style=file -i $filename; done
```
Comment (PetrovKP) [inline .ci/scripts/clang-format.sh]: You need to add **.i** extension. The whole basic realization here.

Comment (napetrov) [reply .ci/scripts/clang-format.sh]: will update it  now

**Pull request merged**